### DISCUSSION TITLE
【lecture 6】关于NTT笔记的第六页

### DISCUSSION THREAD

**Discussion by Delbert-sh** [zkp-co-learning/zkp-co-learn#68] [Q&A（提问题在隔壁~）] [answered]
![image](https://user-images.githubusercontent.com/79981982/226692283-df0cb4a8-cec8-4bbb-b581-8db4969e5085.png)
1.为什么instead of f(1),f(2)...,这里他在干什么？
2.他将coeff form的公式转化为evaluation form之后再转化回来是想用插值吗？

**Comment by Delbert-sh** [comment]
补充一个问题，有没有相关的NTT方面的论文？

**Comment by Howard-Hu** [comment, marked as answer]
推荐FFT的入门资料:https://www.bilibili.com/video/BV1Wh411f71A/

使用 $ f(\omega^0), f(\omega^1),...,f(\omega^{n-1}) $ 来替代f(1),f(2),...,是因为可以方便的递归，可以参考上述视频(视频长度仅半小时)；
FFT将多项式的coeff 形式转化为evaluation 形式；IFFT将多项式的evaluation 形式转化为coeff 形式；
将evaluation形式转化为coeff形式的过程称为插值；

希望我的回答对你有帮助，祝你生活学习愉快
